src/Connection/ServiceConnection.ts

```diff
@@ -466,7 +471,36 @@ export class ServiceConnection {
         const message = JSON.stringify(wrapper);
 
         if (callback) {
-            ConnectionManager.messageReceiver.analyticsSessionRequestCallbacks[requestID] = new ResponseCallback(
+            ConnectionManager.messageReceiver.analyticsRequestCallbacks[requestID] = new ResponseCallback(
+                Date.now(),
+                callback
+            );
+        }
+
+        this.webSocket.send(message);
+    };
+
+    // Function: UpdateAnalyticSessionEvents
+    // Used to send a request to update the analytic session's events stored in the Service
+    UpdateAnalyticSessionEvents = (
```

Comment:
💬 This function and `AnalyticsSessionRequest` are very similar, could we combine them or perhaps make them wrappers around a single generic function?

Reply:
Sounds good - will do that